core/src/main/java/com/datastax/oss/driver/api/core/data/CqlVector.java

```diff
@@ -15,25 +15,31 @@
  */
 package com.datastax.oss.driver.api.core.data;
 
-import com.datastax.oss.driver.shaded.guava.common.base.Joiner;
-import com.datastax.oss.driver.shaded.guava.common.collect.ImmutableList;
-import com.datastax.oss.driver.shaded.guava.common.collect.Iterators;
-import java.util.Arrays;
+import java.util.*;
 
-/** An n-dimensional vector defined in CQL */
-public class CqlVector<T> {
+/**
+ * An n-dimensional vector defined in CQL. You can use either {@link CqlVector#builder()} for an
+ * iterable interface or {@link #CqlVector(List)} directly. Once created, a CqlVector is immutable.
+ */
+public class CqlVector<T> implements Iterable<T> {
 
-  private final ImmutableList<T> values;
+  private final List<T> values;
 
-  private CqlVector(ImmutableList<T> values) {
-    this.values = values;
+  /**
+   * Create a CqlVector from a list of values.
+   *
+   * @param values
+   */
+  public CqlVector(List<T> values) {
+    this.values = Collections.unmodifiableList(values);
   }
 
-  public static Builder builder() {
-    return new Builder();
+  public static <T> CqlVector of(T... values) {
+    return new CqlVector(Collections.unmodifiableList(Arrays.asList(values)));
```

Comment:
This should be newInstance() in order to match up more closely with the other custom types with impls defined in the driver.  CqlDuration is [an excellent example](https://github.com/datastax/java-driver/blob/4.x/core/src/main/java/com/datastax/oss/driver/api/core/data/CqlDuration.java#L96-L108) here.

Reply:
sure, that works too